lucille-core/src/test/java/com/kmwllc/lucille/indexer/SolrIndexerIntegrationTest.java

```diff
@@ -0,0 +1,263 @@
+  }
+
+  @Before
+  public void setup() throws SolrServerException, IOException {
+    cluster.getSolrClient().deleteByQuery(COL, "*:*");
+    cluster.getSolrClient().commit(COL);
+  }
+
+  @Test
+  public void testDeleteById() throws Exception {
+    new UpdateRequest()
+        .add(
+            Arrays.asList(
+                new SolrInputDocument(
+                    "id", "id_1",
+                    "delete_by_id", "deleteid_1")))
+        .setAction(AbstractUpdateRequest.ACTION.COMMIT, true, true)
+        .process(cluster.getSolrClient(), COL);
+
+
+
+    JettySolrRunner jetty = cluster.getJettySolrRunners().get(0);
+
+    Map<String, Object> map = new HashMap<>();
+    map.put("solr.url", jetty.getBaseUrl().toString());
+    map.put("indexer.batchTimeout", 5000);
+    map.put("indexer.batchSize", 1);
+    map.put("indexer.type", "solr");
+    map.put("indexer.sendEnabled", true);
+    map.put("indexer.indexOverrideField", "collection");
+    map.put("indexer.deletionMarkerField", "is_deleted");
+    map.put("indexer.deletionMarkerFieldValue", "true");
+
+    Config config = ConfigFactory.parseMap(map);
+
+    SolrIndexer indexer = null;
+    IndexerMessageManager mockIndexerMessageManager = mock(IndexerMessageManager.class);
+
+    try {
+      indexer = new SolrIndexer(config, mockIndexerMessageManager, false, "solr");
+      assertTrue(indexer.validateConnection());
+
+
+      SolrQuery qr = new SolrQuery();
+      qr.set("q", "*:*");
+
+      assertEquals("The test document should have been indexed.", 1, cluster.getSolrClient().query(COL, qr).getResults().size());
+
+      Document delete = Document.create("id_1");
+      delete.setField("is_deleted", "true");
+      delete.setField("collection", COL);
+      indexer.sendToIndex(List.of(delete));
+
+      cluster.getSolrClient().commit(COL);
+      assertEquals("The test document should have been deleted", 0, cluster.getSolrClient().query(COL, qr).getResults().size());
+
+    } finally {
+      if (indexer != null) {
+        indexer.closeConnection();
+      }
+    }
+  }
+
+  @Test
+  public void testDeleteByField() throws Exception {
+    new UpdateRequest()
+        .add(
+            Arrays.asList(
+                new SolrInputDocument(
+                    "id", "id_1",
+                    "delete_by_id", "deleteid_1")))
+        .setAction(AbstractUpdateRequest.ACTION.COMMIT, true, true)
+        .process(cluster.getSolrClient(), COL);
+
+    JettySolrRunner jetty = cluster.getJettySolrRunners().get(0);
+
+    Map<String, Object> map = new HashMap<>();
+    map.put("solr.url", jetty.getBaseUrl().toString());
+    map.put("indexer.batchTimeout", 5000);
+    map.put("indexer.batchSize", 1);
+    map.put("indexer.type", "solr");
+    map.put("indexer.sendEnabled", true);
+    map.put("indexer.indexOverrideField", "collection");
+    map.put("indexer.deletionMarkerField", "is_deleted");
```

Comment:
A comment about the intent of the test setup would be helpful for future readers of this code -- it takes some time to interpret this.

Suggesting more readable would be:

deletionMarkerField=is_delete
deletionMarkerFieldVlaue=true
deleteByFieldField=my_field
deleteByFieldValue=123

Maybe it's just me but is_deleted being past tense is confusing, and having values of deleteByFieldField and deleteByFieldValue begin with "delete_by_" makes it a little tricker to parse